### PR TITLE
Fix product summary css, hover on mobile and default schema props

### LINKED ISSUE #15: Hide listPrice was equaled to sellingPrice
## Expected Behavior
If `listPrice` was equaled to `sellingPrice` the Price component must hide them. 

## Current Behavior
Isn't been hide.

screen shot 2018-04-18 at 13 52 03

### LINKED ISSUE #18: Hover don't exist in mobile screens
When show button on hover is active on the desktop, on mobile, we need to make the button stay show.

### LINKED ISSUE #23: Set default values in the ProductSummary Schema
The schema properties that have default value equal to `true`, must have a default value in the `ProductSummary.schema.properties`. Otherwise the ComponentEditor will display the value as  `false` 

```javascript
ProductSummary.schema = {
  title: 'Product Summary',
  description: 'The product summary showing the main product informations',
  type: 'object',
  properties: {
    showListPrice: {
      type: 'boolean',
      title: "Show product's list price",
      default: true // <- missing
    },
    ...
```

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

1. Fix the Product Summary CSS (details on screenshots)
2. Fix #15 (Hide list price if equals selling price)
3. Fix #18 (Hover doesn't exist on mobile and tablets)
4. Fix #23 (Default Schema props)

#### What problem is this solving?

Fix Summary issues and it's styling

#### How should this be manually tested?
[Access workspace](https://estacio--storecomponents.myvtex.com/) and see the styling and the Schema props. Access from a smartphone or tablet to see the buy button if it's not hidden completely.

#### Screenshots or example usage
image

Summary specification: 
image

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductName.js [hunk 1]
```diff
@@ -10,7 +10,12 @@ class ProductName extends Component {
 
     return (
       <div className="vtex-product-name overflow-hidden">
-        {referenceCode} {name} {skuName} {brandName && `(${brandName})`}
+        <div className="f5 truncate">
+          {referenceCode} {name}
```
Comment (brunojdo) [inline react/ProductName.js]: Title: productName & branName (Optional)
Subtitle: SKU Name

Comment (estacioneto) [reply react/ProductName.js]: Retirei o ReferenceCode então.

File: react/ProductSummary.js [hunk 2]
```diff
@@ -46,14 +47,14 @@ class ProductSummary extends Component {
       badgeText,
       buyButtonText,
       hideBuyButton,
-      showButtonOnHover,
     } = this.props
 
     const product = this.props.product || createProduct()
+    const showButtonOnHover = this.props.showButtonOnHover && !isMobile
```
Comment (leticiawanderley) [inline react/ProductSummary.js]: I don't think it is necessary to add a dependency for this. You can try to use the media query classes, for example: `this.props.showButtonOnHover ? 'db' : 'db dn-ns'`

File: react/Price.js [hunk 3]
```diff
@@ -67,9 +71,10 @@ class Price extends Component {
                   }}
                 />
               ) : (
-                <span>{installments}X {formattedInstallmentPrice}</span>
-              )
-              }
+                <span>
+                  {installments}X {formattedInstallmentPrice}
```
Comment (leticiawanderley) [inline react/Price.js]: Try to use the unicode value for `times` https://stackoverflow.com/questions/12009545/what-is-the-proper-html-entity-for-the-x-in-a-dimension

File: react/ProductSummary.js (hunk 2 above)
Comment (estacioneto) [reply react/ProductSummary.js]: The thing is that tablets screen sizes are not that small. That's why I wanted to do a mobile detection. Should I ignore the tablets fact?

File: react/Price.js (hunk 3 above)
Comment (estacioneto) [reply react/Price.js]: I made the change, but what about the locales messages? They will be different

Comment (leticiawanderley) [reply react/Price.js]: You can pass the span element containing the Unicode as a parameter to the locale text.

File: react/ProductSummary.js (hunk 2 above)
Comment (leticiawanderley) [reply react/ProductSummary.js]: I still think this is an overkill but have no better solution, so leave it the way it is.

**Pull request merged**